java/fury-core/src/main/java/org/apache/fury/meta/MetaStringDecoder.java

```diff
@@ -74,7 +74,7 @@ private String decodeLowerSpecial(byte[] data) {
     boolean stripLastChar = (data[0] & 0x80) != 0; // Check the first bit of the first byte
     int bitMask = 0b11111; // 5 bits for the mask
     int bitIndex = 1; // Start from the second bit
-    while (bitIndex + 5 <= totalBits) {
+    while (bitIndex + 5 <= totalBits && !(stripLastChar && (bitIndex + 2 * 5 > totalBits))) {
```

Comment:
Could you explain what `!(stripLastChar && (bitIndex + 2 * 5 > totalBits))` mean? It seems not intutive to me.

Reply:
`(bitIndex + 2 * 5 > totalBits)` indicates that the current character is the last character.

`!(stripLastChar && (bitIndex + 2 * 5 > totalBits))` means that if the current character is the last character and the last character needs to be deleted, the while loop will not be entered.

This can reduce the last redundant `StringBuilder#append -> StringBuilder#deleteCharAt`

Reply:
Got it, thanks